provider_test.go

```diff
@@ -58,7 +58,7 @@ func TestGetAPI(test *testing.T) {
 	// when the address is an empty string, we get a nullAPI
 	var api client.API
 
-	api, err := getAPI("", 1)
+	api, err := getAPI("null", "endpoint", 1, "etcd_prefix")
```

Comment:
One of the comments on that linked thread:
"""
The workaround for this problem is to explicitly split the problem into two steps: make one Terraform config that creates the EC2 instance and produces the instance IP address as an output, publish the state from that configuration somewhere, and then use the terraform_remote_state resource to reference that from a separate downstream config that sets up the Docker resources.
"""
So that's what I'm doing currently, but I'll do that expected functionality. I didn't understand why it was operating that way but it makes sense. I'll keep up on that ticket as well.